lib/pure/random.nim

```diff
@@ -368,13 +368,16 @@ proc rand*[T: SomeInteger](t: typedesc[T]): T =
   ##   that accepts a slice
   runnableExamples:
     randomize(567)
+    type E = enum a, b, c, d
     if false: # implementation defined
-      assert rand(int8) == -42
-      assert rand(uint32) == 578980729'u32
-      assert rand(range[1..16]) == 11
+      assert rand(E) in a..d
+      assert rand(char) in low(char)..high(char)
+      assert rand(int8) in low(int8)..high(int8)
+      assert rand(uint32) in low(uint32)..high(uint32)
+      assert rand(range[1..16]) in 1..16
   # pending csources >= 1.4.0 or fixing https://github.com/timotheecour/Nim/issues/251#issuecomment-831599772,
   # use `runnableExamples("-r:off")` instead of `if false`
-  when T is range:
+  when T is range or T is enum or T is char or T is bool:
```

Comment:
imho, only range and enum should be in this branch. For char it's fine to cast and bool needs another procedure like `ratio` that returns true in some given ratio like 1/n or n/m.

Reply:
Indeed char does work cast, I misinterpreted the test results.
I think it's fine to leave `bool`, `rand(bool)` would be equivalent to `ratio` with equal probabilities.